src/Application/Tools/Algorithm/fibheap.cc

```diff
@@ -0,0 +1,656 @@
+#define _FIBHEAP_CPP
+
+//***************************************************************************
+// This Fibonacci heap implementation is Copyright (c) 1996 by John Boyer.
```

Comment:
Is this lifted strait or is it modified?  we should include a link to the repo: https://github.com/ljzhu/FastGrowCut.  If you changed it enough, we should change the copyright.

Reply:
I didn't change this code.

Reply:
Should I add our copyright?

Reply:
no, just a link